Makefile

```
@@ -116,47 +119,100 @@ undeploy: kustomize
 	$(KUSTOMIZE) build config/default$(SUFFIX) | kubectl delete --ignore-not-found -f -
 
 
-# Generate manifests-gen e.g. CRD, RBAC etc.
-manifests-gen: controller-gen
-	$(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
-
+##@ custom targets
 
-manifests$(SUFFIX): manifests-gen kustomize configure
-	cd $@; rm -f *.yaml
-	cd $@; ( $(KUSTOMIZE) build ../config/namespace && echo "---" && $(KUSTOMIZE) build ../config/default$(SUFFIX) ) | $(CSPLIT)
-	cd $@; bash ../scripts/rename.sh
-	cd $@; $(KUSTOMIZE) build ../config/cr > 0016_specialresource_special-resource-preamble.yaml
-
-# Run go fmt against code
-fmt:
-	go fmt ./...
+MANIFEST_DIR = manifests$(SUFFIX)
 
-# Run go vet against code
-vet:
-	go vet --mod=vendor ./...
-
-# Generate code
-generate: controller-gen
-	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
-
-# Build the docker image
-local-image-build: patch helm-lint helm-repo-index test generate manifests-gen
-	podman build -f Dockerfile.ubi8 --no-cache . -t $(IMAGE)
-
-# Push the docker image
-local-image-push:
-	podman push $(IMAGE)
+# Populate manifests dir, and SRO specific customizations
+manifests-gen: manifests kustomize configure
+	rm -f $(MANIFEST_DIR)/*.yaml
+	cd $(MANIFEST_DIR); ( $(KUSTOMIZE) build ../config/namespace && echo "---" && $(KUSTOMIZE) build ../config/default$(SUFFIX) ) | $(CSPLIT)
+	cd $(MANIFEST_DIR); bash ../scripts/rename.sh
+	cd $(MANIFEST_DIR); $(KUSTOMIZE) build ../config/cr > 0017_specialresource_special-resource-preamble.yaml
```

I saw that you increased this by 1 (0016->0017), but I think this file should have been 0017 already, before the addition of `manifests/0002_serviceaccount_special-resource-controller-manager.yaml`. Can you replace the `0017_specialresource_special-resource-preamble.yaml` file with `0018_specialresource_special-resource-preamble.yaml`
```suggestion
	cd $(MANIFEST_DIR); $(KUSTOMIZE) build ../config/cr > 0018_specialresource_special-resource-preamble.yaml
```

---

Well spotted @dagrayvid, thanks. I refactored this target so that all resources are first written to a bundle file and then split. This should prevent such overwrites in the future. PTAL.